apps/api/src/domain/models/video/commands/translate-video-name/translate-video-name.command-handler.ts

```diff
@@ -0,0 +1,36 @@
+import { CommandHandler, ICommand } from '@coscrad/commands';
+import { Valid } from '../../../../../domain/domainModelValidators/Valid';
+import { DeluxeInMemoryStore } from '../../../../../domain/types/DeluxeInMemoryStore';
+import { InMemorySnapshot } from '../../../../../domain/types/ResourceType';
+import { InternalError } from '../../../../../lib/errors/InternalError';
+import { ResultOrError } from '../../../../../types/ResultOrError';
+import { Video } from '../../../audio-item/entities/video.entity';
+import { BaseUpdateCommandHandler } from '../../../shared/command-handlers/base-update-command-handler';
+import { BaseEvent } from '../../../shared/events/base-event.entity';
+import { TranslateVideoName } from './translate-video-name.command';
+import { videoNameTranslated } from './video-name-translated.event';
+
+@CommandHandler(TranslateVideoName)
+export class TranslateVideoNameCommandHandler extends BaseUpdateCommandHandler<Video> {
+    protected async fetchRequiredExternalState(_: TranslateVideoName): Promise<InMemorySnapshot> {
+        return Promise.resolve(new DeluxeInMemoryStore({}).fetchFullSnapshotInLegacyFormat());
+    }
+
+    protected validateExternalState(
+        _state: InMemorySnapshot,
+        _instance: Video
+    ): InternalError | Valid {
+        return Valid;
+    }
+
+    protected actOnInstance(
+        Video: Video,
+        { text, languageCode }: TranslateVideoName
+    ): ResultOrError<Video> {
+        return Video(text, languageCode);
```

Comment:
Note that you should never call a class constructor without the `new` keyword. But you don't want to call the class constructor here anyway.